base/vulkanexamplebase.cpp

```diff
@@ -684,11 +684,52 @@ void VulkanExampleBase::initVulkan(bool enableValidation)
 		vkTools::exitFatal("Could not enumerate phyiscal devices : \n" + vkTools::errorString(err), "Fatal error");
 	}
 
-	// Note :
-	// This example will always use the first physical device reported,
-	// change the vector index if you have multiple Vulkan devices installed
-	// and want to use another one
-	physicalDevice = physicalDevices[0];
+	if (gpuCount == 1)
+	{
+		deviceSelected = 0;
+	}
+	else if (deviceSelected < 0 || deviceSelected >= gpuCount)
+	{
+		std::cout << "Available devices:";
+		for (size_t i = 0; i < gpuCount; i++)
+		{
+			vkGetPhysicalDeviceProperties(physicalDevices[i], &deviceProperties);
+			std::cout << i << "\t(";
+			switch (deviceProperties.deviceType)
+			{
+				case VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU:
+					std::cout << "Integrated GPU";
+					break;
+				case VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU:
+					std::cout << "Discrete GPU";
+					break;
+				case VK_PHYSICAL_DEVICE_TYPE_VIRTUAL_GPU:
+					std::cout << "Virtual GPU";
+					break;
+				case VK_PHYSICAL_DEVICE_TYPE_CPU:
+					std::cout << "CPU";
+					break;
+				case VK_PHYSICAL_DEVICE_TYPE_OTHER:
+					std::cout << "Unknown type";
+					break;
+			}
+			std::cout << ") " << deviceProperties.deviceName;
+			std::cout << std::endl;
+		}
+
+#if 1
+		// Use the first physical device reported if no valid value is provided
+		deviceSelected = 0;
+		std::cout << "Using #" << deviceSelected << std::endl;
+		std::cout << "Choose an other device from the list with `--gpu`" << std::endl;
+#else
+		// Require user to choose a device
+		vkTools::exitFatal("Please choose a device from the list with `--gpu`");
+#endif
```

Comment:
Forgot to ask: which of these two do you prefer?
On invalid/missing user selection, if there's more than one device:
- select first device
- print an error and abort

I'm leaning towards the former (enabled here), as users are notorious for never reading error messages and would open issues instead of just providing the argument requested.